torch/nn/parallel/data_parallel.py

```diff
@@ -26,6 +26,11 @@ class DataParallel(Module):
     other types will be a shallow copy and can be corrupted if written to in
     the model's forward pass.
 
+    .. warning::
+        Forward and backwrad hooks defined on :attr:`module` and its submodules
+        won't be invoked anymore, unless the hooks are initialized in the
+        :meth:`forward` method.
```

Comment:
Why this warning? I don't think that's the case. IIRC the hooks will be called in every replica

Reply:
I actually experimented a bit. Unfortunately it's not the case. The replicas are created using `module.__new__(type(module))` so nothing is attached to them.

Reply:
Hi,
is it possible to somehow overcome this limitation? Maybe some sort of hack?